tests/tests.py

```diff
@@ -0,0 +1,1026 @@
+
+        self.assertEqual(matrix_to_xyz_rpy(modified_joint.origin)[2],
+                         2)
+
+    def test_add_position_fails_if_no_dimension(self):
+
+        modifier = JointModifier.from_name('aligned_link_joint_after', self.modified_robot)
+
+        modification = Modification()
+        modification.add_position(2, absolute=True)
+
+        with self.assertRaises(Exception) as context:
+            modifier.modify(modification)
+
+        self.assertTrue('Axis not specified for joint' in str(context.exception))
+
+class FixedOffsetModifierTests(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(FixedOffsetModifierTests, self).__init__(*args, **kwargs)
+        self.original_filename = './test_model.urdf'
+        self.original_robot, _ = utils.load_robot_and_gazebo_plugins(self.original_filename, 'dummy.urdf')
+
+    def setUp(self):
+        self.modified_robot = copy.deepcopy(self.original_robot)
+
+    def test_calculate_offset_function(self):
+
+        modifier = FixedOffsetModifier.from_name('aligned_link', self.modified_robot)
+
+        parent_joint_offset, child_joint_offset = modifier.calculate_offsets()
+
+        self.assertEqual(parent_joint_offset, 0.5)
+        self.assertEqual(child_joint_offset, -0.5)
+
+    def test_relative_modifier_dimension_for_aligned_joint(self):
+
+        modifier = FixedOffsetModifier.from_name('aligned_link', self.modified_robot)
+
+        modification = Modification()
+        modification.add_dimension(2, absolute=False)
+
+        parent_joint_offset, child_joint_offset = modifier.calculate_offsets()
+
+        modifier.modify(modification)
+
+        modified_parent_joint_offset, modified_child_joint_offset = modifier.calculate_offsets()
+
+        original_link = [link for link in self.original_robot.links if link.name == 'aligned_link'][0]
+        modified_link = [link for link in self.modified_robot.links if link.name == 'aligned_link'][0]
+
+        original_parent_joint = [joint for joint in self.original_robot.joints if joint.child == 'aligned_link'][0]
+        modified_parent_joint = [joint for joint in self.modified_robot.joints if joint.child == 'aligned_link'][0]
+
+        self.assertEqual(modified_link.visuals[0].geometry.cylinder.length,
+                         original_link.visuals[0].geometry.cylinder.length * 2)
+
+        self.assertEqual(matrix_to_xyz_rpy(original_parent_joint.origin)[0],
+                         matrix_to_xyz_rpy(modified_parent_joint.origin)[0])
+
+        self.assertEqual(matrix_to_xyz_rpy(original_parent_joint.origin)[1],
+                         matrix_to_xyz_rpy(modified_parent_joint.origin)[1])
+
+        self.assertEqual(matrix_to_xyz_rpy(original_parent_joint.origin)[2],
+                         matrix_to_xyz_rpy(modified_parent_joint.origin)[2])
+
+        self.assertEqual(modified_parent_joint_offset,
+                         parent_joint_offset)
+
+        self.assertEqual(modified_child_joint_offset,
```

Comment:
Just FYI for comparison of doubles tipically [`self.assertAlmostEqual`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual) is used, as slightly different operations on floating point numbers can give different (even if very similar) results. If `assertEqual`  works in this case it is fine to use it, but notice that it is possible that it stops working in slightly different conditions.

Reply:
For the tests we did this was working, but possibly because we used quite "stable" numbers (that didn't have any rounding issues). I would say for now it is not critical, but if we see we have such issues in the future, for the tests, we can change it, thanks for the suggestion!